test/remote/gateways/remote_ixopay_test.rb

```diff
@@ -5,36 +5,43 @@ def setup
     @gateway = IxopayGateway.new(fixtures(:ixopay))
 
     @amount = 100
-    @credit_card = credit_card('4000100011112224')
+    @credit_card = credit_card('4111111111111111')
     @declined_card = credit_card('4000300011112220')
+
     @options = {
       billing_address: address,
-      description: 'Store Purchase'
+      shipping_address: address,
+      email: 'test@example.com',
+      description: 'Store Purchase',
+      ip: '192.168.1.1',
     }
   end
 
   def test_successful_purchase
     response = @gateway.purchase(@amount, @credit_card, @options)
+
     assert_success response
-    assert_equal 'REPLACE WITH SUCCESS MESSAGE', response.message
-  end
+    assert_equal 'FINISHED', response.message
+    assert_match(/[0-9a-zA-Z]+(|[0-9a-zA-Z]+)*/, response.authorization)
 
-  def test_successful_purchase_with_more_options
-    options = {
-      order_id: '1',
-      ip: "127.0.0.1",
-      email: "joe@example.com"
-    }
+    assert_equal @credit_card.name,           response.params['card_holder']
+    assert_equal '%02d' % @credit_card.month, response.params['expiry_month']
+    assert_equal @credit_card.year.to_s,      response.params['expiry_year']
+    assert_equal @credit_card.number[0..5],   response.params['first_six_digits']
+    assert_equal 'FINISHED',                  response.params['return_type']
 
-    response = @gateway.purchase(@amount, @credit_card, options)
-    assert_success response
-    assert_equal 'REPLACE WITH SUCCESS MESSAGE', response.message
+    assert_equal @credit_card.number.split(//).last(4).join, response.params['last_four_digits']
+
+    assert_not_nil response.params['purchase_id']
+    assert_not_nil response.params['reference_id']
   end
 
   def test_failed_purchase
-    response = @gateway.purchase(@amount, @declined_card, @options)
+    response = @gateway.purchase(@amount, @declined_card, {})
+
     assert_failure response
-    assert_equal 'REPLACE WITH FAILED PURCHASE MESSAGE', response.message
+    assert_equal 'The transaction was declined', response.message
+    assert_equal '2003', response.params['code']
```

Comment:
Should this be checking `response.error_code`?

Reply:
It could be. The response ```error_code``` is a field that we extract from the response and name like so. ```response.params['code']``` is the raw code before extraction. I think I had been focusing the remote tests on verifying that the remote server returned the expected data, whereas the corresponding unit test checks the extracted field ```error_code```. We could do both here though.

Reply:
The line above this checks `response.message`, which is something that was extracted out from the response. I figured the intent of these remote tests was to ensure that the response that we're returning is what we expect it to be, which would include testing the `response_from_request_action` method (and therefore the `error_code_from` method).

Reply:
Added that additional assertion.

Reply:
Why would this be an additional assertion instead of replacing `response.params['code']`? What is the value in checking both within the same test? 

I thought that the translation of `response.params['code']` to `response.error_code` was already being tested in the unit tests here: https://github.com/activemerchant/active_merchant/pull/3396/files?file-filters%5B%5D=.rb#diff-332f420dc6b50a19dc1f41508e903506R37

Reply:
Removed the initial assertion.